app-next/components/Chat.jsx

```diff
@@ -1,13 +1,12 @@
 "use client";
 import React, { useState } from "react";
 
-const Chat = () => {
-  const [messages, setMessages] = useState([]);
+const Chat = ({ messages, onSendMessage }) => {
```

Comment:
You can receive `initialMessages` here, and initialize `messages` state with it, and then move the event handler from parent page to here.